### DISCUSSION TITLE
Show off: blog post about how you can build your own personal cloud!

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16820] [Show and tell]
Hello Cockpit developers and users,

I want to show off a project of mine: I wrote a very lengthy blog post about how you can build your own personal cloud using Red Hat Enterprise Linux, Cockpit, Apache, Nextcloud HUB and Collabora Online.

Your can find the blog post here:
<strike>https://jeroenverhoeckx.com/build-your-own-online-ide.html</strike>
https://jeroenverhoeckx.com/build-your-own-personal-cloud.html

I believe this is one of the most comprehensive blog posts about how you can build such a kind of server!
Cockpit is used for setting up and running the virtual machines.

If you like it, please share it somewhere (for example on Twitter)!

Yours sincerely,

Jeroen Verhoeckx

ps. Thanks for you help and thinking along (a few months ago)!

**Comment by martinpitt** [comment]
Hey @Verhoeckx, thanks for documenting this!

Your blog still says it's a draft -- this should probably be removed before announcing it?

Some thoughts:
 - The "goal" could be a bit more verbose -- like, "self-hosting", links to the top-level projects (e.g. I have no idea what Gitpod and penpot are; I do know nextcloud, but a link couldn't hurt either); also, explaining in short what they do. E.g. it's not obvious to me why you'd want/need nextcloud for an online IDE.
 - VM Installation starts with "log into cockpit", but it should explain how to install it first -- link to https://cockpit-project.org/running.htm
 - It then says "Click on the storage pool you created earlier", but there's no such instruction
 - Not clear where to download the RHEL image from -- this requires a subscription, and later on you suggest that it's enough to get/use a free developer subscription. Rearrange this a bit so that getting the subscription comes first?
 - Should use RHEL 8.5 now instead of 8.4?
 - Could use a pointer to Lavarel homepage/documentation and a quick intro what it is.

**Reply by Verhoeckx** [reply]
_Oh, damn, I shared the wrong url :disappointed: !_

It should be:
https://jeroenverhoeckx.com/build-your-own-personal-cloud.html

The blog post about building your own online IDE is indeed still a draft.
Your points are valid!

Hopefully you like the other blog post more :smiley: (it's very comprehensive)!

**Comment by martinpitt** [comment]
Oh my, that is an awesome blog post indeed, great work! I tweeted it here: https://twitter.com/CockpitProject/status/1482369900597370881

**Reply by Verhoeckx** [reply]
_Great that you like it :sweat_smile: !_

Thanks for sharing it on Twitter :smiley: !!

Could you add the address of the blog to the tweet?

**Reply by Verhoeckx** [reply]
Oh, and I can't finish the other blog about the online IDE because of a possible bug in Apache (the rewrites directives don't seem the work correctly for websockets).

**Reply by martinpitt** [reply]
How embarassing, sorry! :cry:  Deleted the old tweet, and did this one: https://twitter.com/CockpitProject/status/1482383652512309250

**Reply by Verhoeckx** [reply]
No problem, my error (sharing the wrong url) was bigger :smiley: !

_Thank you very much for sharing!!_
